packages/twenty-front/src/modules/views/hooks/useUpdateViewAggregate.ts

```diff
@@ -1,77 +1,91 @@
+import { coreViewsState } from '@/views/states/coreViewState';
+import { convertCoreViewToView } from '@/views/utils/convertCoreViewToView';
+import { produce } from 'immer';
+import { useRecoilCallback } from 'recoil';
+import { findById, isDefined } from 'twenty-shared/utils';
+import { type CoreView } from '~/generated/graphql';
 
 export const useUpdateViewAggregate = () => {
   const { canPersistChanges } = useCanPersistViewChanges();
   const currentViewId = useRecoilComponentValue(
     contextStoreCurrentViewIdComponentState,
   );
   const { updateView } = usePersistView();
+  const { loadRecordIndexStates } = useLoadRecordIndexStates();
 
-  const setRecordIndexKanbanAggregateOperationState = useSetRecoilState(
-    recordIndexKanbanAggregateOperationState,
-  );
+  const updateViewAggregate = useRecoilCallback(
+    ({ set }) =>
+      async ({
+        kanbanAggregateOperationFieldMetadataId,
+        kanbanAggregateOperation,
+        objectMetadataItem,
+      }: {
+        kanbanAggregateOperationFieldMetadataId: string | null;
+        kanbanAggregateOperation: ExtendedAggregateOperations | null;
+        objectMetadataItem: ObjectMetadataItem;
+      }) => {
+        if (!canPersistChanges) {
+          return;
+        }
 
-  const { refreshCoreViewsByObjectMetadataId } =
-    useRefreshCoreViewsByObjectMetadataId();
+        const convertedKanbanAggregateOperation = isDefined(
+          kanbanAggregateOperation,
+        )
+          ? convertExtendedAggregateOperationToAggregateOperation(
+              kanbanAggregateOperation,
+            )
+          : null;
 
-  const updateViewAggregate = useCallback(
-    ({
-      kanbanAggregateOperationFieldMetadataId,
-      kanbanAggregateOperation,
-      objectMetadataId,
-    }: {
-      kanbanAggregateOperationFieldMetadataId: string | null;
-      kanbanAggregateOperation: ExtendedAggregateOperations | null;
-      objectMetadataId: string;
-    }) => {
-      if (!canPersistChanges) {
-        return;
-      }
+        if (!isDefined(currentViewId)) {
+          return;
+        }
 
-      const convertedKanbanAggregateOperation = isDefined(
-        kanbanAggregateOperation,
-      )
-        ? convertExtendedAggregateOperationToAggregateOperation(
-            kanbanAggregateOperation,
-          )
-        : null;
+        const updatedViewResult = await updateView({
+          id: currentViewId,
+          input: {
+            kanbanAggregateOperationFieldMetadataId,
+            kanbanAggregateOperation: convertedKanbanAggregateOperation,
+          },
+        });
+
+        if (updatedViewResult.status === 'successful') {
+          const updatedCoreView = updatedViewResult.response.data
+            ?.updateCoreView as CoreView;
+
+          if (!isDefined(updatedCoreView)) {
+            return;
+          }
+
+          set(coreViewsState, (currentCoreViews) =>
+            produce(currentCoreViews, (draftCoreViews) => {
```

Comment:
a bit exotic?

Reply:
Ok right, I'll create an array util for this use case which is quite common.